### PR TITLE
Increase testability and test coverage of JS code - preliminary

### PR DESCRIPTION
**Pull request by wkurniawan07**
Not fixing any particular issue, but this is to set the stage for #6085.

Although mostly cleaning up (quite a handful of unused codes, duplicate codes, and unnecessary scripts), this will give a bit of sneak-peek on what issues targeted for #6085 will revolve around:
- Getting rid of `onclick` attribute from HTML templates - replace with delegated events instead, expect to see more of this
- Using the pattern used in BootboxWrapper (use an object to contain the functions related to it, much like jQuery) to administrator.js - expect to see this in many more of the scripts
- Moved `$(document).ready` up for better maintainability

### CODE REVIEW

**Comment by wkurniawan07** [comment]
@junhaoyap let's review this?

**Comment by junhaoyap** [comment]
Yep - just got back from a holiday, give me a day

**Comment by junhaoyap** [comment]
This is probably going to be harder to look from the changes so I will look commit by commit

File: src/main/webapp/js/student.js [hunk 1]
```diff
@@ -4,9 +4,9 @@
  */
 
 $(document).ready(function() {
-    $('[data-unreg].navLinks').click(function(event) {
-        event.preventDefault();
-        $clickedLink = $(event.target);
+    $('[data-unreg].navLinks').click(function(e) {
+        e.preventDefault();
```
Comment (junhaoyap) [inline src/main/webapp/js/student.js]: I believe this is more of a matter of preference

**Comment by junhaoyap** [comment]
8d1224b - OK
463d42c - OK
65f0e04 - I can understand the CSS part, I'm more interested to know how we went about detecting that the sortTable part is useless
b45a543 - OK, is it possible for us to add a lint for JS to prevent that?
d648f77 - OK
99f741b - OK, I like the standardization too
9090490 - OK
58dabf4 - OK
8aeeb8b - OK, we might want to take a look at our bad "habit" of preventing all defaults and returning false in the future

Great stuff in this PR generally, just have some questions - we can merge this in the meantime

**Comment by wkurniawan07** [comment]
@damithc this is ready for review.

**Comment by damithc** [comment]
If possible, add header comments for new functions. Can merge after that.

**Comment by wkurniawan07** [comment]
@damithc I did not write any new function; only migrating, in some parts. I aim to add documentations, along with tests and cleaning up, slice-by-slice.

**Comment by wkurniawan07** [comment]
@junhaoyap apologies for missing the questions asked. Anyway, my two cents:

> Is it also possible to add a lint for declared but never used stuff?

I'm afraid not.

> Is it possible for us to add a lint for JS to prevent that?

It is, but not right now, because IIRC the same rule will also lint all our dependencies to globally-defined functions in other files (essentially the namespace-polluting functions). I hope to have this rule activated by the end of next few months, though, i.e after the PR "Increase testability and test coverage of JS code" closes for good.

> We might want to take a look at our bad "habit" of preventing all defaults and returning false in the future.

Agree, and I would need your advice and help on my subsequent PRs that tidy up the JS files.

**Pull request merged**